remix-lib/src/execution/execution-context.js

```diff
@@ -279,18 +279,20 @@ function ExecutionContext () {
     } else {
       web3.setProvider(new web3.providers.HttpProvider(endpoint))
     }
-    if (web3.isConnected()) {
-      executionContext = context
-      self._updateBlockGasLimit()
-      self.event.trigger('contextChanged', ['web3'])
-      self.event.trigger('web3EndpointChanged')
-      cb()
-    } else {
-      web3.setProvider(oldProvider)
-      var alertMsg = 'Not possible to connect to the Web3 provider. '
-      alertMsg += 'Make sure the provider is running and a connection is open (via IPC or RPC).'
-      cb(alertMsg)
-    }
+    web3.eth.net.isListening((err, isConnected) => {
+      if (!err && isConnected) {
+        executionContext = context
+        self._updateBlockGasLimit()
```

Comment:
`self` is alias of `this` here.

Reply:
`self` was used because when you run a closure with `function` keyword, the `this` object is changed, so you need to create a alias for that.
Here you're using `() => ` arrow function that keeps the scrope of the `this` object. So it's better to use `this` than `self`.